src/components/intro/InduceSignup/index.test.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { createMemoryHistory } from 'history';
+import { Router, MemoryRouter } from 'react-router-dom';
+
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import InduceSignup from '.';
+
+describe('<InduceSignup/> component test', () => {
+  it('회원가입을 유도할 수 있는 문구를 렌더링해야 합니다.', () => {
+    const title = 'Go to sign up quickly';
+    const description = 'Get user feedback with Telbby';
+
+    render(
+      <MemoryRouter>
+        <InduceSignup />
+      </MemoryRouter>,
+    );
+
+    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(title);
+    expect(screen.queryByText(description)).toBeInTheDocument();
+  });
+
+  it('"GO"를 클릭하게 되면 회원가입 페이지로 이동해야 합니다.', () => {
+    const history = createMemoryHistory();
+
+    render(
+      <Router history={history}>
+        <InduceSignup />
+      </Router>,
+    );
+
+    fireEvent.click(screen.getByText('GO'));
+
+    expect(history.location.pathname).toBe('/signup');
```

Comment:
`href` 속성에 `/signup` value가 있음을 테스트하는 것으로 대체해도 괜찮지 않을까요?

```suggestion
    render(
      <MemoryRouter>
        <InduceSignup />
      </MemoryRouter>,
    );

    expect(screen.getByRole('link')).toHaveAttribute('href', '/signup');
```

Reply:
link에 해당하는 태그에 href 속성으로 `/signup`이 지정되어 있다고 해서 해당 페이지로 이동한다는 걸 보장할 수 없다고 생각했습니다. `e.preventDefault();`를 사용하면 링크를 막을 수 있기도 하고요.

추가적인 내용으로 react testing library를 사용할 땐 보통 사용자가 애플리케이션을 이용하는 관점에서 테스트를 작성하는 `Behavior Driven Test(행위 주도 테스트) 방법론`을 따릅니다.

 사용자가 원하는 동작은 페이지 이동이기에 최종적으로 페이지가 이동되는 것을 테스트를 해야 하지, href가 /signup으로 지정된 것을 테스트해서는 안된다고 생각합니당. 그건 세부 구현사항을 검증하는 `Implementation Driven Test(구현 주도 테스트)`에서 사용되는 방식이라 보여집니다. 

링크로 구현되든, 버튼으로 onClick 받아서 history.push로 페이지를 이동시키든, 클릭했을 때 페이지가 이동되는 것을 확인하는 것이 해당 케이스의 목적이지만 이런 목적에 부합하지 않는 방식이라고 생각합니다.

Reply:
다만 getByText 대신 getByRole을 사용하는 건 좋은 것 같네요!

Reply:
오... 좋습니다 BDD 에 대한 키워드도 알아가네요 :)

Reply:
오 !! 저는 네브바 컴포넌트 구현할 때 이 부분 history로 test 구현하려다가 실패했는데 성공하셨군요 !!
앞으로 이 코드를 참고해서 테스트 하겠습니당 👍🏻
행위 주도 테스트 방법론, BDD에 대해서도 알게 되었네요! 자세한 설명 감사합니다!!!